lldb/source/Symbol/SwiftASTContext.cpp

```diff
@@ -2034,39 +2034,49 @@ lldb::TypeSystemSP SwiftASTContext::CreateInstance(lldb::LanguageType language,
 
   // Now if the user fully specified the triple, let that override the one
   // we got from executable's options:
-  if (target.GetArchitecture().IsFullySpecifiedTriple()) {
-    swift_ast_sp->SetTriple(target.GetArchitecture().GetTriple());
-  } else {
-    bool set_triple = false;
-    PlatformSP platform_sp(target.GetPlatform());
+  {
+    auto get_executable_triple = [&]() -> llvm::Triple {
+      if (exe_module_sp)
+        if (auto ts = exe_module_sp->GetTypeSystemForLanguage(
+                lldb::eLanguageTypeSwift))
+          if (auto *exe_ast_ctx = llvm::dyn_cast_or_null<SwiftASTContext>(&*ts))
+            return exe_ast_ctx->GetLanguageOptions().Target;
+      return {};
+    };
+
+    llvm::Triple computed_triple;
     llvm::Triple target_triple = target.GetArchitecture().GetTriple();
-    if (platform_sp && !target_triple.hasEnvironment()) {
-      llvm::VersionTuple version =
-          platform_sp->GetOSVersion(target.GetProcessSP().get());
-      std::string buffer;
-      llvm::raw_string_ostream(buffer)
-          << target_triple.getArchName() << '-' << target_triple.getVendorName()
-          << '-' << llvm::Triple::getOSTypeName(target_triple.getOS())
-          << version.getAsString();
-      swift_ast_sp->SetTriple(llvm::Triple(buffer));
-      set_triple = true;
-    }
 
-    if (!set_triple) {
-      auto type_system_or_err =
-          exe_module_sp->GetTypeSystemForLanguage(lldb::eLanguageTypeSwift);
-      if (!type_system_or_err) {
-        llvm::consumeError(type_system_or_err.takeError());
-        return TypeSystemSP();
+    if (target.GetArchitecture().IsFullySpecifiedTriple()) {
+      computed_triple = target_triple;
+    } else {
+      PlatformSP platform_sp(target.GetPlatform());
+      if (platform_sp && !target_triple.hasEnvironment()) {
+        llvm::VersionTuple version =
+            platform_sp->GetOSVersion(target.GetProcessSP().get());
+        std::string buffer;
+        llvm::raw_string_ostream(buffer)
+            << target_triple.getArchName() << '-'
+            << target_triple.getVendorName() << '-'
+            << llvm::Triple::getOSTypeName(target_triple.getOS())
+            << version.getAsString();
+        computed_triple = llvm::Triple(buffer);
+      } else {
+        computed_triple = get_executable_triple();
       }
+    }
 
-      if (ModuleSP exe_module_sp = target.GetExecutableModule()) {
-        auto *exe_swift_ctx =
-            llvm::dyn_cast_or_null<SwiftASTContext>(&*type_system_or_err);
-        if (exe_swift_ctx)
-          swift_ast_sp->SetTriple(exe_swift_ctx->GetLanguageOptions().Target);
-      }
+    if (computed_triple.getOS() == llvm::Triple::MacOSX) {
+      // Handle the case where an apparent macOS binary has been
+      // force-loaded as a macCatalyst process. The Xcode test
+      // runner works this way.
```

Comment:
So the target is macOS but the executable is macCatalyst?

Reply:
This is really tricky:
- the Mach-O binary is `macosx`
- the _target_ it also `macosx` because LLDB is incorrectly not using debugserver to query the triple <rdar://problem/67281791>
- the executable _image_ is `ios-macabi`

fixing the second point would be the _right_ way to fix this, but this is a rather large and risky refactoring.